module/support/optimisation/NSGA2Optimiser/src/nsga2/Individual.cpp

```diff
@@ -0,0 +1,259 @@
+/* Adapted from https://github.com/dojeda/nsga2-cpp
```

Comment:
Can we install from source via Docker to get the code as a package instead?

Reply:
it's not directly used afaik, it's just adapted from, and the repo has a lot more in it than what we use when I looked.